### PR TITLE
Update alias documentation

### LINKED ISSUE #1681: Alias documentation
# Question / Suggestion

Hi Samuel, thanks for your work on the great library! I had a thought about the documentation regarding `alias`.

The [schema documentation section on field customization](https://pydantic-docs.helpmanual.io/usage/schema/#field-customisation) describes alias as:
> alias: the public name of the field

However, I find this somewhat ambiguous. My first impression upon reading the above was this:
```python
class Foo(BaseModel):
    private_name: int = Field(0, alias="public_name")

foo = Foo()

# I thought that this would work, though it does not:
foo.public_name

# I wasn't sure which of these would work:
foo2 = Foo(public_name=1)
foo3 = Foo(private_name=1)
```

I'm not suggesting changing the behavior, but simply clarifying the documentation. Something along the lines of:
> alias: The name which maps to a field during population/initialization.

Or even just adding a **Note** block below to clarify what "public name" means here (namely, that it's _not_ the name used to _access_ the field.

Thanks!

### PR DESCRIPTION
**Pull request by targhs**
## Change Summary
Add usage example for schema field customization alias

Update documentation verbose and add example for using alias

## Related issue number
fix #1681

## Checklist

* [ ] Unit tests for the changes exist
* [ ] Tests pass on CI and coverage remains at 100%
* [x] Documentation reflects the changes where applicable
* [x] `changes/<pull request or issue id>-<github username>.md` file added describing change
  (see [changes/README.md](https://github.com/samuelcolvin/pydantic/blob/master/changes/README.md) for details)
* [x] My PR is ready to review, **please add a comment including the phrase "please review" to assign reviewers**

### CODE REVIEW

**Comment by targhs** [comment]
As its a documentation change. Test cases aren't required.

**Comment by targhs** [comment]
please review

File: docs/usage/schema.md [hunk 1]
```diff
@@ -61,7 +61,12 @@ It has the following arguments:
 * `default_factory`: a zero-argument callable that will be called when a default value is needed for this field.
     Among other purposes, this can be used to set dynamic default values.
     It is forbidden to set both `default` and `default_factory`.
-* `alias`: the public name of the field
+* `alias`: the public name of the field. The name maps to a field during initialization.
```
Comment (samuelcolvin) [inline docs/usage/schema.md:64]: this will break the bullet points, we should add a whole new section (probably not in `schema.py`) to document alias.

**Review by samuelcolvin — COMMENTED** [review]
This is definitely needed, thank you.

Needs some changes:
* needs a whole new section to describe `alias`
* we need to add docs on the effect of alias with `.dict()` and `.json()`

**Comment by samuelcolvin** [comment]
please update.

**Comment by targhs** [comment]
Need your input here.
1. Needs a whole new section to describe
If not in schema, where? It feels more natural to me if its there along with [](https://pydantic-docs.helpmanual.io/usage/schema/#unenforced-field-constraints)  and [](https://pydantic-docs.helpmanual.io/usage/schema/#modifying-schema-in-custom-fields)
2. Gotcha

**Comment by samuelcolvin** [comment]
> If not in schema, where?

Sorry for the slow reply, in `models.md`. Documentation will be rearranged in V2, but that's the best place for now.

**Comment by samuelcolvin** [comment]
### Notice

See [twitter :bird:](https://twitter.com/pydantic/status/1560554679473303552), I've you'd like this to be included in V1.10, please fix it and request a review **TODAY**.

Or if you need this in V1.10 but don't have time to complete it (or aren't the author), please comment here and on #4324.

**Comment by samuelcolvin** [comment]
No reply, closing.

Let me know if you'd like me to re-open this.

**Pull request closed without merging**